cinema/views.py

```diff
@@ -101,8 +95,27 @@ def get_serializer_class(self):
         if self.action == "retrieve":
             return MovieDetailSerializer
 
+        if self.action == "upload_image":
+            return MovieImageSerializer
+
         return MovieSerializer
 
+    @action(
+        methods=["POST"],
+        detail=True,
+        permission_classes=[IsAdminOrIfAuthenticatedReadOnly],
+        url_path="upload-image"
+    )
+    def upload_image(self, request, pk: None):
+        movie = self.get_object()
+        serializer = self.get_serializer(movie, data=request.data)
+
+        if serializer.is_valid(raise_exception=True):
```

Comment:
if condition is unnecesary because the exception will be triggered  and it will never be evaluated as False.
For the same reason `return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)` will never be reached